app/models/proxy.rb

```diff
@@ -240,30 +236,14 @@ def default_production_endpoint
       production_endpoint = proxy.apicast_configuration_driven ? :apicast_production_endpoint : :hosted_proxy_endpoint
       generate(production_endpoint)
     end
-
-    def default_staging_endpoint_apiap
-      default_staging_endpoint
-    end
-
-    def default_production_endpoint_apiap
-      default_production_endpoint
-    end
   end
 
   # @return DeploymentStrategy
   def deployment_strategy
-    strategy = case deployment_option
-               when 'self_managed' then SelfManagedAPIcast
-               when 'hosted' then HostedAPIcast
-               end
-
-    strategy.try!(:new, self)
-  end
-
-  def deployment_strategy_apiap
```

Comment:
This method was introduced because it was needed in a view https://github.com/3scale/porta/commit/f78c868fb83249cba4556ea0cfb6b6e15bed0818. We shouldn't now promote it entirely to be the new behaviour of `Proxy#deployment_strategy`. Instead, I'd ditch `#deployment_strategy_apiap`, keep `#deployment_strategy` the way it is right now (with the case condition) and in the view revert it to how it was used before:

```ruby
proxy.default_staging_endpoint unless proxy.self_managed?
```